src/legacy/timer/timer.go

```diff
@@ -0,0 +1,206 @@
+package timer
+
+import (
+	"unsafe"
+)
+
+type Timer struct {
+	Flags        uint32 // 0, bit 1: raw(set) or interp(unset), bit 2: set when updated
+	Current      uint32 // 4, current value
+	Target       uint32 // 8, target value
+	DeltaPerTick uint32 // 12, (interp only) the amount to add per tick
+	MaxTickDelta uint64 // 16, (interp only) the maximum delta per tick
+	LastUpdated  uint64 // 24, the last tick the value was updated
+}
+
+var _ = [1]struct{}{}[32-unsafe.Sizeof(Timer{})]
+
+func (timer *Timer) C() unsafe.Pointer {
+	return unsafe.Pointer(timer)
+}
+
+type TimerGroup struct {
+	// Timers[0]: usually internal value, ranged from 0 to 0x4000
+	// Timers[1]: For UI display? ranged from 0 to 100
+	// Timers[2]: Not quite sure. Ranged from 0 to 0x2000
+	Timers [3]Timer
+}
+
+var _ = [1]struct{}{}[96-unsafe.Sizeof(TimerGroup{})]
+
+func (timerGroup *TimerGroup) C() unsafe.Pointer {
+	return unsafe.Pointer(timerGroup)
+}
+
+func TimerInit(self *Timer, target int32) int32 {
```

Comment:
```suggestion
func (t *Timer) Init(target int32) int32 {
```
Could make it a method, since its already split from C stuff. Same for other functions that operate on one timer or timer group.

Also note that by Go conventions we do not call the receiver variable `self` or `this` or `timerGroup`. We usually pick a short name like `t` for timer, etc.